nptdms/tdms.py

```diff
@@ -731,6 +808,110 @@ def as_dataframe(self, absolute_time=False):
+                scaled_data = self.data * slope + intercept
+            else:
+                scaled_data = self.data
+            self._data_scaled = scaled_data
+
+        return self._data_scaled
+
+
+class _TdmsmxDAQMetadata(object):
+    __slots__ = [
+        'chunk_size',
+        'data_type',
+        'dimension',
+        'raw_data_widths',
+        'scale_id',
+        'scaler_data_type',
+        'scaler_data_type_code',
+        'scaler_raw_buffer_index',
+        'scaler_raw_buffer_index',
+        'scaler_raw_byte_offset',
+        'scaler_sample_format_bitmap',
+        'scaler_vector_length',
+        ]
+
+    def info(self):
+        l = []
+        for name in self.__slots__:
+            l.append("%s: %s" % (name, getattr(self, name)))
+
+        tmp = ",".join(l)
+        fmt = "%s: ('%s')"
+        txt = fmt % (self.__class__.__name__, tmp)
+        return txt
+
+    def _read_metadata(self, f):
+        """
+        Read the metadata for a DAQmx raw segment.  This is the raw
+        DAQmx-specific portion of the raw data index.
+        """
+        self.data_type = tdsDataTypes[0xFFFFFFFF]
+        self.dimension = _read_long(f)
+        # In TDMS format version 2.0, 1 is the only valid value for dimension
+        if self.dimension != 1:
+            log.warning("Data dimension is not 1")
+        self.chunk_size = _read_long_long(f)
+        # size of vector of format changing scalers
+        self.scaler_vector_length = _read_long(f)
+        #log.debug("mxDAQ Data dimension '%d' n values '%d'"
+        # %(self.dimension, self.chunk_size))
+        # Size of the vector
+        log.debug("mxDAQ format scaler vector size '%d'" %
+                  (self.scaler_vector_length,))
+        if self.scaler_vector_length > 1:
+            log.error("mxDAQ multiple 'format changing' "
+                      "scalers not implemented")
+
+        for idx in range(self.scaler_vector_length):
+            # WARNING: This code overwrites previous values with new
+            # values.  At this time NI provides no documentation on
+            # how to use these scalers and sample TDMS files do not
+            # include more than one of these scalers.
+            self.scaler_data_type_code = _read_long(f)
+            self.scaler_data_type = tdsDataTypes[self.scaler_data_type_code]
+            #log.debug(
+            #    "mxDAQ scaler_vector_length '%d' data_type '%d' = '%s'"
+            #    %(self.scaler_vector_length, self.scaler_data_type_code,
+            # self.scaler_data_type.name))
+
+            # more info for format changing scaler
+            self.scaler_raw_buffer_index = _read_long(f)
+            self.scaler_raw_byte_offset = _read_long(f)
+            self.scaler_sample_format_bitmap = _read_long(f)
+            self.scale_id = _read_long(f)
+
+        #msg = "raw buffer index '%d' raw byte offset '%d' "\
+            #"sample_format_bitmap '%d' scale id '%d'"
+        #log.debug(msg %(
```

Comment:
If these debug log messages might be useful to debug issues later on then please leave them in, otherwise delete them rather than leave commented out code. The log methods also take a format string with the string formatting arguments, so you can just do `log.debug(msg, self.scaler_raw_buffer_index, self.scaler_raw_byte_offset, self.scaler_sample_format_bitmap, self.scale_id)`. This has the benefit that the string interpolation is not computed unless the log level is set to debug and the message will actually be logged.